graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/ast/expressions/AdditionExpression.java

```diff
@@ -92,6 +91,13 @@ public Object evaluateUnsafe(EvaluationContext context) {
             } else {
                 return new Duration(right, left);
             }
+        }else if(String.class.equals(type) ){
+            if(!isPlus){
+                return null;
```

Comment:
That is the correct return value, but I think the type checker should raise an error to the user when it sees this.

The corresponding code is at https://github.com/Graylog2/graylog2-server/blob/eb392d9180e552723a1b9805c6a06f270a16a18d/graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/parser/PipelineRuleParser.java#L777-L780

In that if/else I think it should have a check for `leftType`/`rightType` being `String`, and then in an extra if check for `!isPlus()` which then raises a properly described `InvalidOperation` error. That way the user doesn't get surprised at runtime.

Reply:
Thanks I missed that, I think It's ok now !